clang/test/SemaSYCL/spirv-builtin-lookup.cpp

```diff
@@ -0,0 +1,12 @@
+// RUN: %clang_cc1 -fdeclare-spirv-builtins -fsyntax-only -verify -std=c++11 %s
+// expected-no-diagnostics
+
+// Verify that __spirv_ocl_acos is recognized as a builtin
+
+float acos(float val) {
+  return __spirv_ocl_acos(val);
+}
+
+double acos(double val) {
+  return __spirv_ocl_acos(val);
```

Comment:
Could you also add a CodeGen test, please?

Reply:
Yes, done here https://github.com/intel/llvm/pull/1345/commits/d4bec11b8dbe0610a2066ad6c70460006f6b9bf9